include/bootloader/core/pipette_type.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+typedef enum Type {
+    SINGLE_CHANNEL,
+    EIGHT_CHANNEL,
+    NINETY_SIX_CHANNEL,
+    THREE_EIGHTY_FOUR_CHANNEL
+} PipetteType;
+
+
+#ifdef PIPETTE_TYPE_DEFINE
+
+static inline PipetteType get_pipette_type() { return PIPETTE_TYPE_DEFINE; }
+
+#else
+
+static inline PipetteType get_pipette_type() { return SINGLE_CHANNEL; }
```

Comment:
[nit] Because `PIPETTE_TYPE_DEFINE` exists for every target in cmake, I think a `#error` here makes more sense than having a default value - the only case where we should hit this `#else` is if something is wrong with the build config.

Reply:
`node_id_stm32g4xx.c` which uses `pipette_type.h` is compiled by all the bootloader types whereas only the pipette bootloaders actually specify a `PIPETTE_TYPE_DEFINE`. If we threw an error, all the other bootloaders wouldn't be able to compile successfully. It's kinda gross, but I didn't want to further split out files for this.